Gems/ScriptAutomation/Assets/AutomationScripts/GenericRenderScreenshotTest.lua

```diff
@@ -0,0 +1,56 @@
+----------------------------------------------------------------------------------------------------
+--
+-- Copyright (c) Contributors to the Open 3D Engine Project.
+-- For complete copyright and license terms please see the LICENSE at the root of this distribution.
+--
+-- SPDX-License-Identifier: Apache-2.0 OR MIT
+--
+----------------------------------------------------------------------------------------------------
+function GetRequiredStringValue(valueKey, prettyName)
+    value = g_SettingsRegistry:GetString(valueKey)
+    if (not value:has_value()) then
+        Print('FrameTime script missing ' .. tostring(prettyName) .. ' settings registry entry, ending script early')
+        return false, nil
+    end
+    return true, value:value()
+end
+
+function GetOptionalUIntValue(valueKey, defaultValue)
+    return g_SettingsRegistry:GetUInt(valueKey):value_or(defaultValue)
+end
+function GetOptionalStringValue(valueKey, defaultValue)
+    return g_SettingsRegistry:GetString(valueKey):value_or(defaultValue)
+end
+
+-- required settings
+local LevelPathRegistryKey <const> = "/O3DE/ScriptAutomation/ImageCapture/LevelPath"
+local TestGroupNameRegistryKey <const> = "/O3DE/ScriptAutomation/ImageCapture/TestGroupName" -- used as part of capture filepath, no whitespace or other invalid characters
+local ImageNameRegistryKey <const> = "/O3DE/ScriptAutomation/ImageCapture/ImageName"
+
+-- optional settings
+local ImageComparisonLevelRegistryKey <const> = "/O3DE/ScriptAutomation/ImageCapture/ImageComparisonLevel"
+
+-- check for SettingsRegistry values that must exist
+succeeded, levelPath = GetRequiredStringValue(LevelPathRegistryKey, "Image Capture Level Path")
+if (not succeeded) then return end
+succeeded, testGroupName = GetRequiredStringValue(TestGroupNameRegistryKey, "Test Group Name")
+if (not succeeded) then return end
+succeeded, imageName = GetRequiredStringValue(ImageNameRegistryKey, "Image Capture Name")
+if (not succeeded) then return end
+local imageComparisonLevel = GetOptionalStringValue(ImageComparisonLevelRegistryKey, "Level A") -- default to most strict comparison
+
+
+RunScript("@gemroot:ScriptAutomation@/Assets/AutomationScripts/GenericImageComparisonTestEnvironment.lua")
+captureName = testGroupName .. "/" .. imageName
+
+ExecuteConsoleCommand("r_displayInfo=0")
+ExecuteConsoleCommand("LoadLevel " .. levelPath)
+IdleFrames(2)
+IdleSeconds(5) -- wait for the level to load
```

Comment:
Do we have a script function to wait level load? Can we add one if there isn't?

Reply:
As far as I can tell O3DE has no way to know if a level has finished loading. From the discussions I have seen this is by design.

Reply:
There is a LevelSystemLifecycleNotificationBus sends out an [OnLoadingComplete](https://github.com/o3de/o3de/blob/development/Code/Legacy/CrySystem/LevelSystem/SpawnableLevelSystem.cpp#L487-L488) event when the level loading is complete.

Reply:
We just need to reflect the LevelSystemLifecycleNotificationBus to the BehaviorContext

```c++
behaviorContext->EBus<LevelSystemLifecycleNotificationBus>("LevelSystemLifecycleNotificationBus")
    ->Attribute(AZ::Script::Attributes::Scope, AZ::Script::Attributes::ScopeFlags::Automation)
    ->Attribute(AZ::Script::Attributes::Category, "Level")
    ->Attribute(AZ::Script::Attributes::Module, "level")
    ->Event("OnLoadingComplete", &LevelSystemLifecycleNotificationBus::OnLoadingComplete)
    ->Event("OnLoadingError", &LevelSystemLifecycleNotificationBus::OnLoadingError)
;
```

Reply:
Does the OnLoadingComplete happen after all assets are finished loading? or just the level prefab?

Reply:
I added a LevelLoad function that pauses the script until an `OnLoadingComplete` is received from the `LevelSystemLifecycleNotificationBus`. There still needs to be a small wait after that for textures to finish streaming in, but it achieves most of the requested goals.